src/util/uint.rs

```diff
@@ -403,12 +428,60 @@ macro_rules! construct_uint {
                 Ok($name(ret))
             }
         }
+
+        #[cfg(feature = "serde")]
+        impl $crate::serde::Serialize for $name {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: $crate::serde::Serializer,
+            {
+                use $crate::hashes::hex::ToHex;
+                serializer.serialize_str(&self.to_be_bytes().to_hex())
```

Comment:
This appears to always serialize as a string, even if the output format is binary. This results in suboptimal encodings. We avoid this in [other areas of the code](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/serde_utils.rs#L89) by switching the encoding strategy based on the human readability.

Reply:
Agreed, implemented in https://github.com/rust-bitcoin/rust-bitcoin/pull/511/commits/d0eed01b254f0758fa12c5f84de35993d75c6fb2.

Testing this required introducing a new (dev only) dependency on a serde binary serializer (bincode). I wasn't sure if that's desirable or not so I kept it in a separate commit for now, will squash if its acceptable.